apps/src/dance/SongSelector.tsx

```diff
@@ -0,0 +1,114 @@
+import React, {useState, useCallback, ChangeEvent} from 'react';
+import classnames from 'classnames';
+
+import {commands as audioCommands} from '@cdo/apps/lib/util/audioApi';
+import {getFilteredSongKeys} from '@cdo/apps/dance/songs';
+import Button from '@cdo/apps/templates/Button';
+import {SongData} from '@cdo/apps/dance/types';
+import moduleStyles from '@cdo/apps/dance/song-selector.module.scss';
+
+const commonI18n = require('@cdo/locale');
+
+const currentTimeoutsMap: {[key: string]: ReturnType<typeof setTimeout>} = {};
+const songPreviewDurationInMs = 10000;
+
+interface SongSelectorProps {
+  selectedSong: string;
+  levelIsRunning?: boolean;
+  setSong: (songId: string) => void;
+  songData: SongData;
+  enableSongSelection?: boolean;
+  filterOn?: boolean;
+}
+
+const SongSelector: React.FC<SongSelectorProps> = ({
+  selectedSong,
+  levelIsRunning,
+  setSong,
+  songData,
+  enableSongSelection,
+  filterOn,
+}) => {
+  const [songInPreview, setSongInPreview] = useState(false);
+
+  const onPreviewBtnClick = useCallback(() => {
+    if (songInPreview) {
+      audioCommands.stopSound({url: songData[selectedSong].url});
+      setSongInPreview(false);
```

Comment:
In theory, shouldn't stopping the sound here also trigger the `onEnded` callback and therefore also call `setSongInPreview(false)`? Not saying we should remove the state update here, but if we're doing this here, maybe worth being consistent and also explicitly calling `setSongInPreview(false)` in `changeSong`? What do you think?

Reply:
Nice catch, thank you! Yeah, we indeed don't need `setSongInPreview(false)` in timer since onEnded will be called anyways. +1 to stay consistent. Just removed this line so we now only have one `setSongInPreview(false)` call. (also removed it from line 45)

Reply:
Cool, yeah I was thinking we could also just add `setSongInPreview(false)` to changeSong, but either way works!